server/services/amazonScrapingService.js

```diff
@@ -0,0 +1,50 @@
+const puppeteer = require("puppeteer");
+
+exports.getProductDetail = async (productUrl) => {
+  if (productUrl) {
+    try {
+      const browser = await puppeteer.launch({ headless: true });
+      const page = await browser.newPage();
+      await page.goto(productUrl, { waitUntil: "domcontentloaded" });
+
+      const productData = await page.evaluate(() => {
+        const featuresUlElem = document
+          .querySelector("div#feature-bullets")
+          .querySelector("ul");
+        const featuresLiElem = Array.from(
+          featuresUlElem.querySelectorAll("li")
+        );
+
+        const features = featuresLiElem.map((liElem) => {
+          return liElem.innerText;
+        });
+
+        let price = "unavailable";
+
+        try {
+          price = document.querySelector("span#priceblock_ourprice").innerText;

```

Comment:
I'll check if this works for regular and sale price.

Reply:
It seems that the span#priceblock_ourprice selector is for both regular and sale prices. In fact, this selector is for the current sale price